### PR TITLE
emit typed event for user management

### PR DESCRIPTION
**Pull request by blizzz**
Next to the legacy event, fire a typed event. Needed for impersonate's IBootstrap implementation

### CODE REVIEW

**Comment by blizzz** [comment]
Psalm's complain is unrelated.

**Review by ChristophWurst — APPROVED** [review]

File: apps/settings/lib/Events/LoadAdditionalScriptsEvent.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Settings\Events;
+
+use OCP\EventDispatcher\Event;
+
+class LoadAdditionalScriptsEvent extends Event {
```
Comment (ChristophWurst) [inline apps/settings/lib/Events/LoadAdditionalScriptsEvent.php:30]: though the naming could be improved. how about something similar to the ``\OCA\Files_Sharing\Event\BeforeTemplateRenderedEvent`` event?

Comment (blizzz) [reply apps/settings/lib/Events/LoadAdditionalScriptsEvent.php:30]: or `OCA\Files\Event\LoadAdditionalScriptsEvent`? :)

Comment (blizzz) [reply apps/settings/lib/Events/LoadAdditionalScriptsEvent.php:30]: though your suggestions makes more sense, i agree

Comment (ChristophWurst) [reply apps/settings/lib/Events/LoadAdditionalScriptsEvent.php:30]: yeah the idea was that the event name should describe what's happening and not what to do :)

**Comment by blizzz** [comment]
should be mentioned then in https://github.com/nextcloud/server/issues/20953#issue-617327468 as well

**Review by kesselb — APPROVED** [review]
LGTM :+1:

**Comment by skjnldsv** [comment]
> LGTM

ci says no

**Comment by MorrisJobke** [comment]
#14552

**Pull request merged**